### PR TITLE
Add codecov to travis.yml

### LINKED ISSUE #582: Add CodeCov support to .travis.yml
Issue Summary

Add this to our travis.ci configuration so that code coverage tests will run!
https://github.com/codecov/example-node/blob/master/.travis.yml

### PR DESCRIPTION
**Pull request by troglotit**
# Fixes #582

### Checklist
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the master branch.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation about the functionality in the appropriate .md file
- [ ] I have added in line documentation to the code I modified

### Short description of what this PR does:
- Adds codecov to travis.yml
- This PR is duplicate of #586, but I thought this would pass CI.

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-nodejs?pullRequest=598) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-nodejs?pullRequest=598) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-nodejs?pullRequest=598) it.</sub>

**Comment by mbernier** [comment]
@troglotit

We have not been able to merge your Pull Request, but because you are awesome - we wanted to make sure you could still get a SendGrid Hacktoberfest shirt.  

Please go **fill [out our swag form](https://dx.sendgrid.com/swag) before Nov 5th** and we will send the shirt! (We know that you might have tried this before and it didn’t work, sorry about that!)

### **You have till Nov 5th to fill out this form** in order to get the Hacktoberfest shirt!

Thank you for contributing during Hacktoberfest! We hope to see you in the repos soon! Just so you know, we always give away a SendGrid shirt for your first ever non-Hacktoberfest PR that gets merged.

File: .travis.yml [hunk 1]
```diff
@@ -19,6 +19,8 @@ script:
 - yarn test:helpers
 - yarn test:typescript
 - yarn test:license
+- ./node_modules/.bin/istanbul cover ./node_modules/mocha/bin/_mocha --reporter lcovonly -- -R spec
```
Comment (nitish24p) [inline .travis.yml]: You can move this to a scripts command in package json just to maintain uniformity..

Comment (nitish24p) [reply .travis.yml]: Just a suggestion

**Comment by troglotit** [comment]
@nitish24p thank you for your review! I've addressed your comments, seems totally reasonable.

Check are not successful, because of `yarn test:files.spec` - I've pulled from master, but it's red anyway.

**Comment by nitish24p** [comment]
@troglotit  yea don't worry. Travis is failing on the master branch. I've made a PR which will fix that. So no need to worry.

**Pull request closed without merging**